### PR TITLE
Enable crate sorting with `rustfmt`

### PR DESCRIPTION
**Pull request by charlesbluca**
This PR builds off of #817 to enable crate sorting through `rustfmt`, a feature currently only contained in nightly packages.

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,3 +31,12 @@ repos:
...
       - id: check-added-large-files
+  - repo: local
+    hooks:
+      - id: cargo-fmt
+        name: cargo fmt
+        description: Format files with cargo fmt.
+        entry: cargo +nightly fmt
+        language: system
+        types: [rust]
+        args: ['--manifest-path', './dask_planner/Cargo.toml', '--verbose', '--']
```
Comment (ayushdg) [inline .pre-commit-config.yaml:42]: I'm not too familiar with the syntax but does this imply users should have the nightly cargo-fmt tool setup already?

**Review by ayushdg — APPROVED** [review]
Small question but lgtm! Thanks for this @charlesbluca

File: .pre-commit-config.yaml (hunk 1 above)
Comment (charlesbluca) [reply .pre-commit-config.yaml:42]: It does - thanks for the reminder, I will add a blurb to `CONTRIBUTING.md` explaining how to install the nightly toolchain to run the pre-commit hooks properly

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,12 +4,13 @@
...
 
-- a system installation of [`rustup`](https://rustup.rs/)
+- a system installation of [`rustup`](https://rustup.rs/) with the latest stable and nightly toolchains
 - a [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) environment containing all required Python packages
 
-Once `rustup` is installed, ensure that the latest stable toolchain is available by running
+Once `rustup` is installed, ensure that the latest stable and nightly toolchains are available by running
 
 ```
+rustup toolchain install nightly
```
Comment (charlesbluca) [inline CONTRIBUTING.md:13]: IIUC, this should install all components of the nightly Rust toolchain, though we should only need `rustfmt`. Is there a command we can use to _only_ install nightly `rustfmt`?

cc @andygrove

Comment (ayushdg) [reply CONTRIBUTING.md:13]:
```
rustup component add rustfmt --toolchain nightly
cargo +nightly fmt
```
Taken from https://github.com/rust-lang/rustfmt#on-the-nightly-toolchain

**Comment by ayushdg** [comment]
rerun tests

**Comment by charlesbluca** [comment]
rerun tests

**Pull request merged**